lib/v2/newzmz/index.js

```diff
@@ -1,90 +1,62 @@
 const got = require('@/utils/got');
 const cheerio = require('cheerio');
-const { art } = require('@/utils/render');
-const path = require('path');
+
+const { rootUrl, getItems, getItemInfo, processItems } = require('./util');
 
 module.exports = async (ctx) => {
-    const category = ctx.params.category ? parseInt(ctx.params.category) : 1;
+    const { id = '1', downLinkType = '磁力链' } = ctx.params;
+    const limit = ctx.query.limit ? parseInt(ctx.query.limit, 10) : 5;
 
-    const rootUrl = 'http://newzmz.com';
-    const rootSUrl = 'http://s.newzmz.com';
-    const currentUrl = `${rootUrl}/index.html`;
+    // If the id is not composed solely of digits,
+    // then consider it as the id of a movie or TV show;
+    // otherwise, consider it as the id for the category.
 
-    const response = await got({
-        method: 'get',
-        url: currentUrl,
-    });
+    const isCategory = !isNaN(id);
 
-    const $ = cheerio.load(response.data);
+    const currentUrl = new URL(isCategory ? 'index.html' : `details-${id}.html`, rootUrl).href;
 
-    const items = [];
+    const { data: response } = await got(currentUrl);
```

Comment:
https://github.com/DIYgod/RSSHub/actions/runs/5878489009/job/15940677110#step:12:16
https://github.com/DIYgod/RSSHub/actions/runs/5878489009/job/15940677110#step:12:29

The same URL is still being visited twice.